module/purpose/Defend/src/Defend.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Defend.hpp"
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+
+#include "message/localisation/Ball.hpp"
+#include "message/localisation/Field.hpp"
+#include "message/purpose/Player.hpp"
+#include "message/strategy/WalkToFieldPosition.hpp"
+#include "message/support/FieldDescription.hpp"
+
+#include "utility/math/euler.hpp"
+
+namespace module::purpose {
+
+    using extension::Configuration;
+
+    using DefendMsg = message::purpose::Defend;
+
+    using message::localisation::Ball;
+    using message::localisation::Field;
+    using message::strategy::WalkToFieldPosition;
+    using message::support::FieldDescription;
+    using utility::math::euler::pos_rpy_to_transform;
+
+    Defend::Defend(std::unique_ptr<NUClear::Environment> environment) : BehaviourReactor(std::move(environment)) {
+
+        on<Configuration>("Defend.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file Defend.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+        });
+
+        on<Provide<DefendMsg>, With<Ball>, With<Field>, With<FieldDescription>>().then(
```

Comment:
looks good, but quick question, whats the expected behavior of this when the ball in between the robot and our own goal, will it always be expected to switch to an attacking purpose?

Reply:
Yeah the idea is it should just become the attacker if the ball is coming for it. Only issue I've seen is when the attacking robot is chasing the ball down so it remains the attacking robot. Added logic on the attacking robot could help, or reducing the equidistance value.